### PR TITLE
Split Screen Case Search: Dynamically update results

### PR DESCRIPTION
**Pull request by stephherbers**
## Product Description

Main build PR for [this ticket](https://dimagi-dev.atlassian.net/browse/USH-3544) but builds off https://github.com/dimagi/commcare-hq/pull/33428 that was combined in this PR

dynamic (1)

Note in the gif the first results are not filtered for DOB since Name is a required field, once that is filled out the results are filtered after field change. 

## Technical Summary

If the FF is enabled and the sidebar, then on field change it checks if all required fields are non-empty, and if so then perfrom a search. 

## Feature Flag

DYNAMICALLY_UPDATE_SEARCH_RESULTS and SPLIT_SCREEN_CASE_SEARCH
### Safety story

This will be tested on staging including for performance implications. Our biggest concern is the number of requests added if a search is performed after each field change. This is FF'ed separately from SSCS so if problems arise, the previous behavior of SSCS can be enabled without requiring a deploy immediately. 

### Automated test coverage

None

### QA Plan

Planning on dev testing on staging

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/toggles/__init__.py [hunk 1]
```diff
@@ -1030,14 +1030,23 @@ def _ensure_valid_randomness(randomness):
...
+
 SPLIT_SCREEN_CASE_SEARCH = StaticToggle(
     'split_screen_case_search',
     "Split screen case search: In case search, show the search filters in a sidebar on the left and the results"
     " on the right.",
     TAG_CUSTOM,
     help_link='https://confluence.dimagi.com/display/USH/Split+Screen+Case+Search',
     namespaces=[NAMESPACE_DOMAIN],
-    parent_toggles=[SYNC_SEARCH_CASE_CLAIM]
+    parent_toggles=[SYNC_SEARCH_CASE_CLAIM, DYNAMICALLY_UPDATE_SEARCH_RESULTS]
```
Comment (snopoke) [inline corehq/toggles/__init__.py:1049]: should this not be inverted? I think `DYNAMICALLY_UPDATE_SEARCH_RESULTS` is a 'child' of `SPLIT_SCREEN_CASE_SEARCH`

**Review by nospame — COMMENTED** [review]
It seems like the first two commits are part of https://github.com/dimagi/commcare-hq/pull/33428/ and this PR just depends on them, is that correct?

Also, there is a ticket to add testing for this feature -- it seems like tests should be added to this PR before it gets merged into master.

In general, though, these changes look good.  (Reviewed as a pair with @jingcheng16)

File: corehq/toggles/__init__.py (hunk 1 above)
Comment (stephherbers) [reply corehq/toggles/__init__.py:1049]: This is what I originally had put, but for the functionality I wanted through testing that if `SPLIT_SCREEN_CASE_SEARCH` was enabled I wanted `DYNAMICALLY_UPDATE_SEARCH_RESULTS` to be enabled.-- my understanding is that this is the relationship that I would need for that-- is that a correct assumption?

**Comment by stephherbers** [comment]
hey @nospame (and @jingcheng16) thank you for the review-- for the testing ticket I don't know of much automated tests we can add and rather it would be manual testing done on staging including performance

File: corehq/toggles/__init__.py (hunk 1 above)
Comment (snopoke) [reply corehq/toggles/__init__.py:1049]: Sure, that is correct - but it means that you could enable `DYNAMICALLY_UPDATE_SEARCH_RESULTS` without having `SPLIT_SCREEN_CASE_SEARCH` enabled.

If you always want `DYNAMICALLY_UPDATE_SEARCH_RESULTS` enabled with `SPLIT_SCREEN_CASE_SEARCH` it seems like maybe you should just use the same toggle?

**Comment by nospame** [comment]
> hey @nospame (and @jingcheng16) thank you for the review-- for the testing ticket I don't know of much automated tests we can add and rather it would be manual testing done on staging including performance

That seems fair. We don't have a great way or really examples of automated testing for JS views, which I'm seeing is where these changes are. I do think we should work on changing that... but that's a whole other can of worms.

**Comment by stephherbers** [comment]
@snopoke @nospame Simon and I spoke and decided the best approach would be to keep the parent-child relationship has `SPLIT_SCREEN_CASE_SEARCH` (parent) and `DYNAMICALLY_UPDATE_SEARCH_RESULTS` (child) and at a later date I will go back and modify the `set` toggle logic to also enable the child when the parent is enabled so they are both enabled when one is enabled, but then the `DYNAMICALLY_UPDATE_SEARCH_RESULTS` can be disabled independently if needed

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js [hunk 2]
```diff
@@ -534,6 +544,19 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
             });
         },
 
+        updateSearchResults: function () {
+            var self = this;
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js:548]: no `const` and `let`?

**Review by nospame — APPROVED** [review]
Will this now conflict with https://github.com/dimagi/commcare-hq/pull/33428 given the toggles code is slightly different?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js (hunk 2 above)
Comment (stephherbers) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js:548]: ah I was following convention of the rest of the file, but i think this is worth going back to modifying for the whole file if we see fit in a separate PR

**Pull request merged**